src/configure/clients/TemplateServiceClientFactory.ts

```diff
@@ -24,5 +24,9 @@ export class TemplateServiceClientFactory {
         return this.client;
     }
 
+    public static resetClient() {
```

Comment:
This approach has one problem - it exposes the reset functionality to outside the factory. And that can be problematic if rest of the code uses reset in an irresponsible way which can result in client becoming useless midway.

Reply:
Al alternative, if I may propose, is to check whether the cached client uses same credential as incoming one. If yes, then use the cached client, if no then refresh the client.
This assumes that we can get credentials from client. if no, then this approach is not going to work.

Reply:
getting credentials from client is not there but i thought of storing PAT as static member in factory and for every getClient comparing the PAT with incoming PAT but this way i am introducing a dependency for example if in future if there is other param as well on which client is getting dependent then that param also we need to take care.

For 1st point, generally we call getClient function to get the client before using it, and it will take care giving proper client , if someone has used resetClient before getclient by mistake. But if some other class will also caching the client after getClient , then if by mistake they call resetClient and used the cached client then it will be an issue. But in that case that will be wrong approach to cache already cache client (which is static as well )

Reply:
> i thought of storing PAT as static member in factory and for every getClient comparing the PAT with incoming PAT but this way i am introducing a dependency for example if in future if there is other param come on which client is getting dependent then that param also we need to take care.

Agree

Reply:
I have one more suggestion. My primary concern is with exposing explicit function to reset client which seems like an unnecessary extra functionality to consumers. 
What consumer wants is ability to get a client which is refreshed (i.e. not the cached one). If so, then the facory's `getClient()` function can have an additional argument `refresh` which will mean that client wants a uncached client,

Reply:
Also, just looking at `getClient` again,  if the client is sending a `githubPatToken` then it implicitly means that it wants a uncached client. And we should honour that. So, IMO a solution would be make `githubPatToken` as optional argument and use cached client if PAT is not specified and refresh the client if it is specified.
Thoughts?

Reply:
Agree bishal, i will make a change to make service credentials and githubPAT as optional arguments ( because which credentials to take depends upon serviceframework option) and if anyone pass both then only we will create a new client and for rest cases we will return cached one or if cache is not there then will throw an exception